src/components/signup/SignupState.js

```diff
@@ -486,11 +488,14 @@ const Signup = ({ navigation }: { navigation: any, screenProps: any }) => {
       setLoading(false)
 
       return true
-    } catch (e) {
-      log.error('New user failure', e.message, e, { dialogShown: true })
-      showSupportDialog(showErrorDialog, hideDialog, navigation.navigate, e.message)
+    } catch (exception) {
+      const { message, serverResponse } = exception
 
-      // showErrorDialog('Something went on our side. Please try again')
+      log.error('New user failure', message, exception, {
+        dialogShown: true,
+        errorCode: serverResponse && 8,
+      })
+      showSupportDialog(showErrorDialog, hideDialog, navigation.navigate, serverResponse ? message : generalError(8))
```

Comment:
we need to check what are the possible server messages i'm not sure we want to show them

Reply:
@sirpy 
There could be such error messages:
- User mobile not verified!
- User email not verified!
- You cannot create more than 1 account with the same credentials
- Failed adding user to mautic
- Failed adding user to w3
- Failed topping user wallet

Also there is some places where we throw system errors:
```
.catch(e => {
          logger.error('updateMarketToken failed', e.message, e, { userRecordWithPII })
          throw e
        })
```

```
.catch(e => {
            logger.error('addUserToWhiteList failed', e.message, e, { userRecord })
            throw e
          })
```

```
.catch(e => {
          logger.error('failed updating new user record', e.message, e, { toUpdateUser })
          throw e
        })
```

I think we can throw the error with some human-readable error messages here too.
For example:
- Something went wrong while trying to fetch market token
- Something went wrong while trying to white-list user
- Something went wrong while trying update user record

Reply:
lets just stick with the generic error  message